tck/src/main/java/org/eclipse/microprofile/opentracing/tck/OpenTracingWildcardNameTests.java

```diff
@@ -0,0 +1,138 @@
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.eclipse.microprofile.opentracing.tck;
+
+import io.opentracing.tag.Tags;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import javax.ws.rs.HttpMethod;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.Status;
+import org.eclipse.microprofile.config.spi.ConfigSource;
+import org.eclipse.microprofile.opentracing.tck.application.WildcardClassService;
+import org.eclipse.microprofile.opentracing.tck.tracer.TestSpan;
+import org.eclipse.microprofile.opentracing.tck.tracer.TestSpanTree;
+import org.eclipse.microprofile.opentracing.tck.tracer.TestSpanTree.TreeNode;
+import org.jboss.arquillian.container.test.api.Deployment;
+import org.jboss.arquillian.container.test.api.RunAsClient;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.testng.annotations.Test;
+
+/**
+ * @author Pavol Loffay
+ */
+public class OpenTracingWildcardNameTests extends OpenTracingClientBaseTests {
+
+    public static class TestConfiguration implements ConfigSource {
+        private Map<String, String> propMap = new HashMap<>();
+
+        {
+            propMap.put("mp.opentracing.server-operation-name-provider", "http-path");
+        }
+
+        @Override
+        public Map<String, String> getProperties() {
+            return propMap;
+        }
+
+        @Override
+        public String getValue(String s) {
+            return propMap.get(s);
+        }
+
+        @Override
+        public String getName() {
+            return this.getClass().getName();
+        }
+    }
+
+    @Deployment
+    public static WebArchive createDeployment() {
+        return OpenTracingBaseTests.createDeployment()
+            .addAsServiceProvider(ConfigSource.class, TestConfiguration.class);
+    }
+
+    @Override
+    protected String getOperationName(String spanKind, String httpMethod, Class<?> clazz, Method method) {
+        if (spanKind.equals(Tags.SPAN_KIND_SERVER)) {
+            StringBuilder operationName = new StringBuilder();
+            Path classPath = clazz.getAnnotation(Path.class);
+            if (classPath != null) {
+                if (!classPath.value().startsWith("/")) {
+                    operationName.append("/");
+                }
+                debug(classPath.value());
+                operationName.append(classPath.value());
+            }
+            if (!classPath.value().endsWith("/")) {
```

Comment:
This needs to be inside the above block - otherwise classPath = null would cause npe.

Reply:
I am not sure if resource class can be defined without path annotation. I will find out

Reply:
It seems it cannot. Such endpoint is not discovered